go/arrow/array/fixedsize_binary.go

```diff
@@ -0,0 +1,45 @@
+package array
+
+import "github.com/apache/arrow/go/arrow"
+
+// A type which represents an immutable sequence of fixed-length binary strings.
+type FixedSizeBinary struct {
+	array
+	byteWidth    int
+	valueOffsets []int32
+	valueBytes   []byte
+}
+
+// NewFixedSizeBinaryData constructs a new fixed-size binary array from data.
+func NewFixedSizeBinaryData(data *Data) *FixedSizeBinary {
```

Comment:
this needs to be wired into `makeArrayFn`:
https://github.com/apache/arrow/blob/0a8654d3060e9f83a33e8bf22984b80e08cadffe/go/arrow/array/array.go#L182

(together with a test "of course" :))